File: go-version/leetcode/LC3000/area_of_max_diagonal.go
package LC3000

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/8/26 上午8:19
 * @FilePath: area_of_max_diagonal
 * @Description: 一次遍历
 */

func areaOfMaxDiagonal(dimensions [][]int) int {
	var maxArea, maxLength int
	for _, dimension := range dimensions {
		area := dimension[0] * dimension[1]
		length := dimension[0]*dimension[0] + dimension[1]*dimension[1]
		if length > maxLength {
			maxLength = length
			maxArea = area
		} else if length == maxLength {
			maxArea = max(maxArea, area)
		}
	}
	return maxArea
}

File: python-version/LC3000/area_of_max_diagonal.py
from typing import List


class Solution:
    @staticmethod
    def areaOfMaxDiagonal(dimensions: List[List[int]]) -> int:
        max_area, max_length = 0, 0
        for dimension in dimensions:
            area = dimension[0] * dimension[1]
            length = dimension[0]**2 + dimension[1]**2
            if length > max_length:
                max_length = length
                max_area = area
            elif length == max_length:
                max_area = max(max_area, area)
        return max_area
